### PR TITLE
Feature/bin lookup in custom card

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
Convert SecuredFields comp to Preact hooks
Add binLookup capabilities to the CustomCard component
Full list of tasks that this PR involved can be found at: COWEB-955

## Tested scenarios
Written new e2e tests
All tests old and new pass

**Fixed issue**:  COWEB-955

### CODE REVIEW

File: packages/lib/src/components/internal/SecuredFields/binLookup/createCardVariantSwitcher.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { BrandObject, DualBrandSelectElement } from '../../../Card/types';
+
+/**
+ * Creates an object used for setting state - that will trigger the rendering of a select element to allow a choice between 2 different card variants
+ * @param brandObjArr - array containing 2 card brands objects
+ */
+export default function createCardVariantSwitcher(brandObjArr: BrandObject[]) {
+    const leadBrand = brandObjArr[0];
+    const subBrand = brandObjArr[1];
```
Comment (pabloai) [inline packages/lib/src/components/internal/SecuredFields/binLookup/createCardVariantSwitcher.ts:9]:
```suggestion
    const [leadBrand, subBrand] = brandObjArr;
```

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    onChange: (data) => void;
+    i18n: Language;
+}
+
+const defaultProps = {
+    onChange: () => {},
+    onError: () => {}
+};
+
+function SecuredFields(props: SecuredFieldsProps) {
```
Comment (marcperez) [inline packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx:18]: Should this be called `SecuredFieldsInput`?

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+     */
+    useEffect(() => {
+        props.onChange({
+            data,
+            valid,
+            errors,
+            isValid: isSfpValid,
+            selectedBrandValue
+        });
+    }, [data, valid, errors, selectedBrandValue]);
```
Comment (pabloai) [inline packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx:87]: Shall `isSfpValid` be included in the dependency array?

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx (hunk 2 above)
Comment (sponglord) [reply packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx:18]: I'm fed up of too many things being called SecuredFields in too many directories of the same name.
So I'm following the pattern of the Card comp: `Card` & `CardInput`

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx (hunk 3 above)
Comment (sponglord) [reply packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx:87]: ??

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx (hunk 2 above)
Comment (sponglord) [reply packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx]: Misunderstood. Have changed the internal name to SecuredFieldsInput

File: packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx (hunk 3 above)
Comment (pabloai) [reply packages/lib/src/components/SecuredFields/SecuredFieldsInput/SecuredFieldsInput.tsx:87]: Ok, just checking!

**Review by marcperez — APPROVED** [review]

**Pull request merged**